asdf/commands/edit.py

```diff
@@ -0,0 +1,589 @@
+    Rewrite an ASDF file for too large edited YAML.  The edited YAML, a pad,
+    the blocks will be rewritten.  A block index will also be rewritten.  If a
+    block index existed in the old file, it will have to be recomputed to
+    because of the larger YAML size and pad, which changes the location of
+    the binary blocks.
+
+    Parameters
+    ----------
+    edited_text : the new YAML text to write out.
+    orig_text : the original YAML text to overwrite.
+    oname : the ASDF file to overwrite.
+    fname : the edit YAML to write to new file.
+    """
+
+    tmp_oname = oname + ".tmp"  # Save as a temp file, in case anything goes wrong.
+    pad_size = 10 * 1000
+    padded_text = add_pad_to_new_text(edited_text, pad_size)
+
+    ifd = open(oname, "r+b")  # Open old ASDF to get binary blocks
+    ifd.seek(len(orig_text))
+
+    ofd = open(tmp_oname, "w+b")  # Open temp file to write
+    ofd.write(padded_text)  # Write edited YAML
+
+    current_location = len(padded_text)
+    block_index = []
+    alloc_loc = 14  # 4 bytes of block ID, 2 blocks of size, 8 blocks into header
+    block_chunk = 2048
+    while True:
+        next_block = get_next_block_header(ifd)
+        if next_block is None:
+            break
+
+        # Get block size on disk
+        alloc = struct.unpack(">Q", next_block[alloc_loc : alloc_loc + 8])[0]
+
+        # Save block location for block index
+        block_index.append(current_location)
+        current_location = current_location + len(next_block) + alloc
+
+        # Copy block
+        ofd.write(next_block)
+        while alloc >= block_chunk:
+            chunk = ifd.read(block_chunk)
+            ofd.write(chunk)
+            alloc -= block_chunk
+        if alloc > 0:
+            chunk = ifd.read(alloc)
+            ofd.write(chunk)
+    ifd.close()
+
+    write_block_index(ofd, block_index)
+    ofd.close()
+
+    # Rename temp file.
+    os.replace(tmp_oname, oname)
+
+    # Output message to user.
+    delim = "*" * 70
+    print(f"\n{delim}")
+    print(f"The text in '{fname}' was too large to simply overwrite the")
+    print(f"text in '{oname}'.  The file '{oname}' was rewritten to")
+    print("accommodate the larger text size.")
+    print(f"Also, added a '\\n' and {pad_size:,} ' ' as a pad for")
+    print(f"the text in '{oname}' to allow for future edits.")
+    print(f"{delim}\n")
+
+
+def save_func(fname, oname):
+    """
+    Checks to makes sure a corresponding ASDF file exists.  This is done by
+        seeing if a file of the same name with '.asdf' as an extension exists.
+    Checks to makes sure fname is a valid YAML file.
+    If the YAML text is smaller than the YAML text in the ASDF file
+        overwrite the YAML in the ASDF file.
+    If the YAML text is smaller than the YAML text in the ASDF file
+        If the file is small, then rewrite file.
+        If the file is large, ask if rewrite is desired.
+
+    Parameters
+    ----------
+    fname : The input YAML file.
```

Comment:
This docstring is missing the formatting updates.

Reply:
I updated this.

Reply:
I have changed all docstrings to conform to this format.